core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -146,64 +147,84 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
     case Some(topic) => Map("topic" -> topic)
   }
 
+  case class MeterWrapper(metricType: String, eventType: String) {
+    @volatile private var lazyMeter: Meter = _
+    private val meterLock = new Object
+
+    def meter(): Meter = {
+      if (lazyMeter == null) {
+        meterLock synchronized {
+          if (lazyMeter == null)
+            lazyMeter = newMeter(metricType, eventType, TimeUnit.SECONDS, tags)
+        }
+      }
+      lazyMeter
```

Comment:
You have to read `lazyMeter` into a local variable before the null check. Otherwise `null` could be returned if  `close` is called at the right time.